s_mp_radix_size_radix_10.c

```diff
@@ -0,0 +1,56 @@
+#include "tommath_private.h"
+#ifdef S_MP_RADIX_SIZE_RADIX_10_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+
+
+#define LTM_RADIX_SIZE_SCALE 64
+#define LTM_RADIX_SIZE_CONST_SHIFT 32
+int s_mp_radix_size_radix_10(const mp_int *a, size_t *size)
```

Comment:
@czurnieden why don't you add a function s_mp_log10 which is called from mp_log and used indirectly bymp_radix_size?

Reply:
I was, and still am, a bit torn between calling that optimization in `mp_log` or in `mp_radix_size`.

Calling it in `mp_radix_size` allows for easy reduction to the restricted radix-set `2,4,8,10,16,32,64` and would also getting rid of the dependency `mp_log` which isn't a very small function.

Calling it in `mp_log` is cleaner if we want to keep the full radix-set.

Mmh&hellip;
\*strokes sesquipedalian  beard\*
I don't know.

Reply:
Well, but you can strip down mp_log by disabling s_mp_log and only enabling the power of two and base 10 versions. We had that discussion in #389. I think it should go to mp_log since it is more general.

Reply:
If you strip down 'mp_log' you don't have a general log-function anymore.

If you strip down 'mp_radix_size' you still have a radix-size function, just not for the small range of radices 2-64, only for the smaller range of powers-of-two and base 10.

You expect of a log-function that it works over the whole range, with no holes in it.

You expect from a radix-size function to work over a very small range, it might even have holes in it. Restricting the string out/input to only a handfull of bases, sometimes down to only two (10 and 16) won't get many complains (vid. e.g.: `printf`).

Reply:
Yes, but we don't get an optimized log function if we add the base10 optimization only to radix_size. We already have other functions with holes in it if configured as such.